lib/ts/utils.ts

```diff
@@ -207,28 +208,62 @@ export function validateTheStructureOfUserInput(
     }
 }
 
-export async function getAPIVersionFromFileIfExists(): Promise<string | null> {
+export async function getDataFromFileIfExists<T>(filePath: string): Promise<T | undefined> {
     try {
-        let apiVersionFromFile = await new Promise<Buffer>((resolve, reject) => {
-            readFile(API_VERSION_FILE_PATH, (err, data) => {
+        let dataFromFile = await new Promise<Buffer>((resolve, reject) => {
+            readFile(filePath, (err, data) => {
                 if (err !== undefined && err !== null) {
                     reject(err);
                 }
                 resolve(data);
             });
         });
-        return apiVersionFromFile.toString();
+        return JSON.parse(dataFromFile.toString());
     } catch (err) {
-        return null;
+        return undefined;
     }
 }
 
-export async function storeAPIVersionInFile(apiVersion: string) {
+export async function storeIntoTempFile(filePath: string, data: any) {
+    try {
+        await createTempDirIfNotAlreadyExists();
```

Comment:
No.. We do not want to create the dir. It should already be there.. if it does not exist, we do nothing.

Reply:
okay

Reply:
You are not checking if the dir exists or not.. If it does not exist, will trying to write to a file in that dir throw an error?

Reply:
Please reply to this @bhumilsarvaiya

Reply:
yes, if the dir doesn't exists, the writeFile function will throw an error